docs/decisions/0005-studio-lms-subdomain-boundaries.rst

```diff
@@ -0,0 +1,178 @@
+experience of a particular set of students.
+
+Furthermore, the mapping of Content Groups to Cohorts is not 1:1. You could
+decide that both on-campus students and alumni get the same content group
+experience, while keeping those Cohorts separate for the purposes of other parts
+of the LMS like forums discussions.
+
+A more future looking example might be the interaction between Open edX
+courseware and third party forum services. The fact that certain units are
+marked as discussable topics might be a Content Authoring decision in Studio,
+while the choice of which forum service those discussions happen in might be a
+Learning decision in the LMS.
+
+
+Studio Content data acts as an input to LMS policy and Learner Experience data.
+-------------------------------------------------------------------------------
+
+As courseware becomes more dynamic, certain concepts in the LMS are becoming
+richer than their equivalent concepts in Studio. In these situations, we should
+think of the data relationship as a one way flow of data from Studio to the LMS.
+The LMS takes Studio data as an input that it can enrich, transform, or override
+as necessary to create the desired student learning experience.
+
+Content scheduling is a good example of this. In the early days of Open edX,
+course teams would set start and due dates for subsections in Studio, and that
+would be the end of it. Today, we have personalized schedules, individual due
+date extensions, and more. The pattern we use to accomplish this is:
+
+* Copy the schedule information from Studio to the LMS at the time a course is
+  published, transforming it into a more easily queryable form in the process.
+* Add additional data models in the LMS to support use cases like individual due
+  date extensions and personalized rescheduling. This is currently handled by
+  the edx_when app, developed in the edx-when repository.
+* Remap field data so that XBlocks in the LMS runtime query this richer data
+  model. Accessing an XBlock's ``start`` or ``due`` attribute in the Studio
+  runtime continues to work with simple key/values in ModuleStore, but the LMS
+  XBlock runtime will fetch those values from edx-when's in-process API.
+
+This approach allows us to add flexibility to the LMS, while preserving
+backwards compatibility with existing XBlock content.
+
+
+LMS data should not flow backward into Studio.
+----------------------------------------------
+
+Since LMS concepts extend Studio ones, we don't want changes to flow backwards
+from the LMS back into Studio. Some reasons:
+
+* There is no guarantee that Studio course runs will be 1:1 with LMS course
+  runs. In fact, one to many mappings of course runs already exist if CCX
+  courses are enabled.
+* A unidirectional data flow makes the system easier to debug and reason about.
+* The OLX import/export process stays much simpler if it doesn't have to
+  consider data that the LMS has added.
+
```

Comment:
Note for possible future revision:

"Studio" here refers to the content authoring functionality of Studio today, and is looking at this from an architectural lens, and not a user experience one. The site that course teams experience as "Studio" might one day include course administration features (lifted out the Instructor Toolbar). But that should still be a _separated_ piece from the content editing portion, as it may apply to different sets of users.

Similarly, "data should not flow backwards" is a system level perspective. It is perfectly fine for Studio to have an iframe into an Insights panel that gives usage stats about a particular piece of content that you're editing–that data resides in Insights and Studio just has an opaque pointer to some kind of integration endpoint. It is _not_ okay for a change in your forum moderators configuration to flow back and be reflected in the OLX export for a course.